### PR TITLE
add more links

### PR DESCRIPTION
**Pull request by crse**

### CODE REVIEW

File: spec/factories/profile_link_sites.rb [hunk 1]
```diff
@@ -185,6 +185,55 @@
...
+    
+    # Pixiv
+    trait :pixiv do
+      validate_find 'A(https?://)?(www.)?(pixiv.net/)?(member.php?id=)?(?<username>[0-9]+)\z'
+      validate_replace 'https://pixiv.net/member.php?id=\k<username>'
+    end
+    
+    # Booth.pm
+    trait :boothpm do
+      validate_find 'A(https://)?(?<username>[a-zA-z0-9]+)(.booth.pm)?\z'
```
Comment (toyhammered) [inline spec/factories/profile_link_sites.rb]: should be `\A` and also make sure it is `(https?://)?`, you forgot the `?` after the `s`. Check all the ones you did please.

File: spec/factories/profile_link_sites.rb [hunk 2]
```diff
@@ -185,6 +185,55 @@
...
+      validate_replace 'http://\k<username>.booth.pm'
+    end
+    
+    # AniDB
+    trait :anidb do
+      validate_find 'A(https://)?(www.)?(anidb.net/)?(up)?(?<username>[0-9]+)\z'
```
Comment (toyhammered) [inline spec/factories/profile_link_sites.rb]: here

File: spec/factories/profile_link_sites.rb [hunk 3]
```diff
@@ -185,6 +185,55 @@
...
+      validate_replace 'https://anidb.net/up\k<username>'
+    end
+    
+    # AniList
+    trait :anilist do
+      validate_find 'A(https://)?(www.)?(anilist.co/)?(?<username>[a-zA-Z0-9]+)\z'
```
Comment (toyhammered) [inline spec/factories/profile_link_sites.rb]: here

File: spec/factories/profile_link_sites.rb (hunk 1 above)
Comment (toyhammered) [inline spec/factories/profile_link_sites.rb]: should be `\A` and also make sure it is `(https?://)?`, you forgot the `?` after the `s`. Check all the ones you did please, I have marked them as **here**.

File: spec/factories/profile_link_sites.rb [hunk 4]
```diff
@@ -185,6 +185,55 @@
...
+      validate_replace 'https://anilist.co/\k<username>'
+    end
+    
+    # VNDB
+    trait :vndb do
+      validate_find 'A(https://)?(www.)?(vndb.org/)?(u)?(?<username>[0-9]+)\z'
```
Comment (toyhammered) [inline spec/factories/profile_link_sites.rb]: here

File: spec/models/profile_link_site_spec.rb [hunk 5]
```diff
@@ -633,5 +633,165 @@
...
       end
     end
+    
+    # MyAnimeList
+    describe 'MyAnimeList' do
+      context 'success' do
+        it 'should return a username' do
+          urls = %w[
+            myanimelist.net/profile/FukuchiChiisaia
+            http://www.myanimelist.net/profile/FukuchiChiisaia
```
Comment (toyhammered) [inline spec/models/profile_link_site_spec.rb]: I know it doesn't really matter but it should be `https`

File: db/seeds/profile_link_sites.yml [hunk 6]
```diff
@@ -138,3 +138,43 @@ trakt:
...
+  validate_replace: 'https://annict.com/\k<username>'
+myfigurecollection:
+  id: 31
+  name: MyFigureCollection
+  validate_find: '\A(https?://)?(www.)?(myfigurecollection.net/)?(profile/)?(?<username>[a-zA-Z0-9]+)\z'
+  validate_replace: 'https://myfigurecollection.net/profile/\k<username>'
+pixiv:
+  id: 32
+  name: Pixiv
+  validate_find: 'A(https?://)?(www.)?(pixiv.net/)?(member.php?id=)?(?<username>[0-9]+)\z'
```
Comment (toyhammered) [inline db/seeds/profile_link_sites.yml]: here needs to be changed

File: db/seeds/profile_link_sites.yml [hunk 7]
```diff
@@ -138,3 +138,43 @@ trakt:
...
+  validate_replace: 'https://pixiv.net/member.php?id=\k<username>'
+boothpm:
+  id: 33
+  name: Booth.pm
+  validate_find: 'A(https://)?(?<username>[a-zA-z0-9]+)(.booth.pm)?\z'
```
Comment (toyhammered) [inline db/seeds/profile_link_sites.yml]: here

File: db/seeds/profile_link_sites.yml [hunk 8]
```diff
@@ -138,3 +138,43 @@ trakt:
...
+  validate_replace: 'http://\k<username>.booth.pm'
+anidb:
+  id: 34
+  name: AniDB
+  validate_find: 'A(https://)?(www.)?(anidb.net/)?(up)?(?<username>[0-9]+)\z'
```
Comment (toyhammered) [inline db/seeds/profile_link_sites.yml]: here

File: db/seeds/profile_link_sites.yml [hunk 9]
```diff
@@ -138,3 +138,43 @@ trakt:
...
+  validate_replace: 'https://anidb.net/up\k<username>'
+anilist:
+  id: 35
+  name: AniList
+  validate_find: 'A(https://)?(www.)?(anilist.co/)?(?<username>[a-zA-Z0-9]+)\z'
```
Comment (toyhammered) [inline db/seeds/profile_link_sites.yml]: here

File: db/seeds/profile_link_sites.yml [hunk 10]
```diff
@@ -138,3 +138,43 @@ trakt:
...
+  validate_replace: 'https://anilist.co/\k<username>'
+vndb:
+  id: 36
+  name: VNDB
+  validate_find: 'A(https://)?(www.)?(vndb.org/)?(u)?(?<username>[0-9]+)\z'
```
Comment (toyhammered) [inline db/seeds/profile_link_sites.yml]: here

**Review by toyhammered — CHANGES_REQUESTED** [review]
Start by correcting all of these and I will check over it again.

File: db/seeds/profile_link_sites.yml [hunk 11]
```diff
@@ -138,3 +138,33 @@ trakt:
...
+myanimelist:
+  id: 29
+  name: MyAnimeList
+  validate_find: '\A(https?://)?(www.)?(myanimelist.net/)?(profile/)?(?<username>[a-zA-Z0-9_\-]+)\z'
+  validate_replace: 'https://myanimelist.net/profile/\k<username>'
+animeplanet:
+  id: 30
+  name: AnimePlanet
+  validate_find: '\A(https?://)?(www.)?(anime-planet.com/)?(users/)?(?<username>[a-zA-Z0-9_\-]+)\z'
+  validate_replace: 'https://anime-planet.com\users\k<username>'
```
Comment (NuckChorris) [inline db/seeds/profile_link_sites.yml:150]: Honestly I don't really like the thought of giving free advertising to AnimePlanet, AniDB, AniList, or MAL (not that MAL needs it from us lmao)

File: db/seeds/profile_link_sites.yml [hunk 12]
```diff
@@ -138,3 +138,33 @@ trakt:
...
+pixiv:
+  id: 31
+  name: Pixiv
+  validate_find: '\A(https?://)?(www.)?(pixiv.me/)?(?<username>[a-zA-Z0-9]+)\z'
+  validate_replace: 'https://pixiv.me\k<username>'
```
Comment (NuckChorris) [inline db/seeds/profile_link_sites.yml:155]: Pixiv is highly welcome :+1:

**Review by NuckChorris — CHANGES_REQUESTED** [review]
I'm fine with VNDB and Pixiv, if you wanna pare it down to those.  The plan is now to eventually convert this system to something where users can put whatever links they want and then we'll pick an icon for it automatically, and _then_ MAL links are easy.  Until then, I think we'd prefer not giving free advertising 😜

File: db/seeds/profile_link_sites.yml (hunk 12 above)
Comment (JoshFabian) [reply db/seeds/profile_link_sites.yml:155]: I would actually prefer we stop adding more links. It's getting incredibly messy. This feature needs refactored.

Comment (NuckChorris) [reply db/seeds/profile_link_sites.yml:155]: Alright then, I'll close this and we can go forward with the refactor instead

**Pull request closed without merging**